cmd/util/cmd/hotstuff-view/cmd/reader.go

```diff
@@ -0,0 +1,39 @@
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/onflow/flow-go/consensus/hotstuff/persister"
+)
+
+type Reader struct {
+	persister *persister.Persister
+}
+
+func NewReader(persister *persister.Persister) *Reader {
+	return &Reader{
+		persister: persister,
+	}
+}
+
+func (r *Reader) SetHotstuffView(view uint64) error {
+	if view == 0 {
+		return fmt.Errorf("hotstuff view is not allowed to set to 0, please specify --view")
+	}
+
```

Comment:
```suggestion

	// ensure we don't set the view lower than the current view, because this may cause this node to violate protocol rules
	currentView, err := r.GetHotstuffView()
	if err != nil {
		return err
	}
	if view < currentView {
		return fmt.Error("hotstuff view cannot be lower than current view (%d)", currentView)
	}

```

Reply:
Good catch